src/blockchain/Blockchain.ts

```diff
@@ -80,47 +126,26 @@ export default class Blockchain implements IBlockchainApi {
     return this.api.rpc.chain.subscribeNewHeads(listener)
   }
 
-  public async submitTx(
+  public async signTx(
     identity: Identity,
     tx: SubmittableExtrinsic
-  ): Promise<SubmittableResult> {
-    const accountAddress = identity.address
-    const nonce = await this.getNonce(accountAddress)
+  ): Promise<SubmittableExtrinsic> {
+    const nonce = await this.getNonce(identity.address)
     const signed: SubmittableExtrinsic = identity.signSubmittableExtrinsic(
       tx,
       nonce.toHex()
     )
-    log.info(`Submitting ${tx.method}`)
-
-    return new Promise<SubmittableResult>((resolve, reject) => {
-      signed
-        .send((result) => {
-          log.info(`Got tx status '${result.status.type}'`)
-
-          if (ErrorHandler.extrinsicFailed(result)) {
-            log.warn(`Extrinsic execution failed`)
-            log.debug(`Transaction detail: ${JSON.stringify(result, null, 2)}`)
-            const extrinsicError: ExtrinsicError =
-              this.errorHandler.getExtrinsicError(result) || ERROR_UNKNOWN
-
-            log.warn(`Extrinsic error ocurred: ${extrinsicError}`)
-            reject(extrinsicError)
-          }
-          if (result.isFinalized) {
-            resolve(result)
-          } else if (result.isError) {
-            reject(
-              new Error(
-                `Transaction failed with status '${result.status.type}'`
-              )
-            )
-          }
-        })
-        .catch((err: Error) => {
-          // just reject with the original tx error from the chain
-          reject(err)
-        })
-    })
+    return signed
+  }
+
+  // TODO: should this be renamed signAndSubmitTx ?
+  public async submitTx(
```

Comment:
Yea, I think this is better. 

`signAndSubmitTx`

Reply:
This could be `signableAndSubmitableTx`

Reply:
in order not to break the old api, I would want to keep this name although its not super descriptive. I have renamed the other function instead to signal that it requires previous signing (unlike this one)